tests/Integration/PermissionsTest.php

```diff
@@ -0,0 +1,369 @@
+	 * @param Parsely_Options_Content_Helper $pch_options The passed options.
+	 */
+	public function assert_current_user_access_to_pch_feature_with_filter(
+		string $feature,
+		$pch_options
+	): void {
+		// Filter set to true.
+		add_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_true' );
+		self::assertTrue(
+			Permissions::current_user_can_use_pch_feature(
+				$feature,
+				$pch_options
+			)
+		);
+		remove_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_true' );
+
+		// Filter set to false.
+		add_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_false' );
+		self::assertFalse(
+			Permissions::current_user_can_use_pch_feature(
+				$feature,
+				$pch_options
+			)
+		);
+		remove_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_false' );
+	}
+
+	/**
+	 * Verifies that the result of current_user_can_use_pch_feature() is false,
+	 * when an invalid feature name is passed.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @covers \Parsely\Permissions::current_user_can_use_pch_feature
+	 * @uses \Parsely\Permissions::build_pch_permissions_settings_array
+	 */
+	public function test_current_user_attempts_to_access_pch_feature_with_invalid_feature_name(): void {
+		$user_allowed = Permissions::build_pch_permissions_settings_array(
+			true,
+			array( 'administrator' )
+		);
+
+		self::assertFalse(
+			Permissions::current_user_can_use_pch_feature(
+				'invalid_feature',
+				$user_allowed
+			)
+		);
+
+		// Filter set to true.
+		add_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_true' );
+		self::assertFalse(
+			Permissions::current_user_can_use_pch_feature(
+				'invalid_feature',
+				$user_allowed
+			)
+		);
+		remove_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_true' );
+
+		// Filter set to false.
+		add_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_false' );
+		self::assertFalse(
+			Permissions::current_user_can_use_pch_feature(
+				'invalid_feature',
+				$user_allowed
+			)
+		);
+		remove_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_false' );
+	}
+
+	/**
+	 * Asserts the result of current_user_can_use_pch_feature(), when the
+	 * options are unset.
+	 *
+	 * @since 3.17.0
+	 *
+	 * @param string                         $feature The feature to check access for.
+	 * @param Parsely_Options_Content_Helper $pch_options The passed options.
+	 */
+	public function assert_current_user_access_to_pch_feature_with_unset_options(
+		string $feature,
+		$pch_options
+	): void {
+		unset( $pch_options[ $feature ] );
+
+		self::assertFalse(
+			Permissions::current_user_can_use_pch_feature(
+				$feature,
+				$pch_options // @phpstan-ignore-line
+			)
+		);
+
+		// Filter set to true.
+		add_filter( 'wp_parsely_current_user_can_use_pch_feature', '__return_true' );
+			self::assertFalse(
+				Permissions::current_user_can_use_pch_feature(
+					$feature,
+					$pch_options // @phpstan-ignore-line
+				)
+			);
```

Comment:
I believe the code indentation here is a bit offset.
```suggestion
		self::assertFalse(
			Permissions::current_user_can_use_pch_feature(
				$feature,
				$pch_options // @phpstan-ignore-line
			)
		);
```

Reply:
Good catch!